assignments/prototype-refactor.js

```diff
@@ -7,3 +7,131 @@ Prototype Refactor
 2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
 
 */
+  
+/*
+  === GameObject ===
+  * createdAt
+  * name
+  * dimensions (These represent the character's size in the video game)
+  * destroy() // prototype method that returns: `${this.name} was removed from the game.`
+*/
+
+class GameObject{
+    constructor(obj) {
+      this.createdAt = obj.createdAt;
+      this.name = obj.name;
+      this.dimensions = obj.dimensions;
+        }
+        destroy(){
+            return `${this.name} was removed from the game.`;
+        }
+    
+    };
+    /*
+          === CharacterStats ===
+          * healthPoints
+          * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+          * should inherit destroy() from GameObject's prototype
+        */
+    
+        class CharacterStats extends GameObject{
+        constructor(obj) {
+        super(obj);
+      this.healthPoints = obj.healthPoints;
+    }
+    
+        takeDamage() {
+      return `${this.name} took damage.`
+        }
+    };
+    
+    /*
+          === Humanoid (Having an appearance or character resembling that of a human.) ===
+          * team
+          * weapons
+          * language
+          * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+          * should inherit destroy() from GameObject through CharacterStats
+          * should inherit takeDamage() from CharacterStats
+        */
+    
+            class Humanoid extends CharacterStats{
+        constructor(obj) {
+        super(obj);
+      
+      this.teams = obj.team;
```

Comment:
Slight typo here: the console log below on line 126 refers to `team` (`swordsman.team`), but you called the property `teams` here, making that console log return `undefined`. Changing either one so that it matches will fix this.